samples/subsys/zbus/dyn_channel/src/main.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+#include "zbus_messages.h"
+
+ZBUS_CHAN_DEFINE(pkt_channel,		   /* Name */
+		 false,			   /* Read only */
+		 struct external_data_msg, /* Message type */
+
+		 NULL,			 /* Validator */
+		 ZBUS_OBSERVERS(filter), /* observers */
+		 ZBUS_MSG_INIT(0)	 /* Initial value {0} */
+);
+
+ZBUS_CHAN_DEFINE(version,	     /* Name */
+		 true,		     /* Read only */
+		 struct version_msg, /* Message type */
+
+		 NULL,						   /* Validator */
+		 ZBUS_OBSERVERS_EMPTY,				   /* observers */
+		 ZBUS_MSG_INIT(.major = 0, .minor = 1, .build = 1) /* Initial value {0} */
+);
+
+ZBUS_CHAN_DEFINE(data_ready,	 /* Name */
+		 false,		 /* Read only */
+		 struct ack_msg, /* Message type */
+
+		 NULL,			   /* Validator */
+		 ZBUS_OBSERVERS(consumer), /* observers */
+		 ZBUS_MSG_INIT(0)	   /* Initial value {0} */
+);
+
+ZBUS_CHAN_DEFINE(ack,		 /* Name */
+		 false,		 /* Read only */
+		 struct ack_msg, /* Message type */
+
+		 NULL,			   /* Validator */
+		 ZBUS_OBSERVERS(producer), /* observers */
+		 ZBUS_MSG_INIT(0)	   /* Initial value {0} */
+);
+
+void filter_cb(struct zbus_channel *chan)
+{
+	struct external_data_msg *chan_message = NULL;
+
+	__ASSERT_NO_MSG(chan == &pkt_channel);
+	chan_message = (struct external_data_msg *)chan->message;
```

Comment:
Is the casting needed? Think you can just use `chan_message = chan->message` since message is a `void *`.